cpp/ql/lib/ext/deallocation/Std.deallocation.model.yml

```diff
@@ -0,0 +1,41 @@
+extensions:
+  - addsTo:
+      pack: codeql/cpp-all
+      extensible: deallocationFunctionModel
+    data:
+      - ["", "", False, "free", "0"]
+      - ["std", "", False, "free", "0"]
```

Comment:
I think we need the `bsl` version of `free` here.

We also need the `std` versions of most of the rest assuming we want exactly identical behaviour to before.  I can see there's some inconsistency between the way we treated allocations and frees in certain libraries, perhaps this is a deliberate refinement?

Reply:
> I think we need the `bsl` version of `free` here.

Good catch. Totally agree.

> We also need the `std` versions of most of the rest assuming we want exactly identical behaviour to before. I can see there's some inconsistency between the way we treated allocations and frees in certain libraries, perhaps this is a deliberate refinement?

Hm, yes. I'm not actually sure why all the Windows specific functions are also specified to be in the `std` namespace. That's not where they live. I'd consider removing the `std` case from these to be a bugfix, but if you're uncomfortable with this I can add the `std` cases here to make it very clear that this preserves any existing behavior (even though I believe the `std` cases for the Microsoft specific functions add 0 additional functions to the model).

Reply:
The missing `bsl::free` has been added in 36d59cef7e75c2aa0192bd8be4d57c5d82ebf410

Reply:
> I'd consider removing the std case from these to be a bugfix, but if you're uncomfortable with this I can add the std cases here to make it very clear that this preserves any existing behavior

No, I'm happy with this change as long as it's deliberate.  It makes sense that none of these should ever be part of `std` apart from the basics (`malloc` etc).